src/edu/csus/ecs/pc2/core/model/Problem.java

```diff
@@ -2002,4 +2048,131 @@ public void setVivaInputValidatorHasBeenRun(boolean vivaInputValidatorHasBeenRun
+     * on the Edit Problem GUI (or via YAML configuration).
+     * 
+     * @return
+     */
+    public int getMemoryLimitMB() {
+        return memoryLimitMB;
+    }
+    
+    /**
+     * Sets the memory limit for this problem. Setting a memory limit of zero indicates "no limit",
+     * meaning that the problem is constrained only by the memory provided by the hardware, the OS,
+     * and the specific language runtime system.
+     * 
+     * Note that setting a memory limit does not automatically imply that such limit is enforced; 
+     * enforcing a memory limit requires selection of a problem sandbox capable of doing that.
+     * (See {@link EditProblemSandboxPane}.)
+     * 
+     * If a value less than zero is passed in the memory limit is set to zero (unlimited).
+     * 
+     * @param memLimitInMB the memory limit for the problem, in MB; must be >= 0, where 0=unlimited.
+     */
+    public void setMemoryLimit(int memLimitInMB) {
+        if (memLimitInMB < 0) {
+            this.memoryLimitMB = 0;
+            //TODO: pass a Log into the Problem constructor so conditions like this can be logged properly.
+//            getLog().warning("Memory limit < 0 specified; setting to 0 (unlimited)");
+        } else {
+            this.memoryLimitMB = memLimitInMB;
+        }
+    }
+
+    /**
+     * Returns a String containing the name of the sandbox program associated with this Problem.
+     * Note that the value returned by this method is only relevant if the value returned by
+     * {@link #getSandboxType()} is something other than {@link SandboxType#NONE}.
+     * 
+     * @return the currently-defined sandbox program name.
+     */
+    public String getSandboxProgramName() {
+        if (sandboxProgramName == null) {
+            sandboxProgramName = Constants.PC2_INTERNAL_SANDBOX_PROGRAM_NAME;
+        }
+        return sandboxProgramName;
+    }
+
+    /**
+     * Sets the name of the sandbox program used by this Problem.
+     * Note that setting a sandbox program name does NOT in and of itself cause the specified sandbox to be
+     * used; the Admin must configure/enable the sandbox using the Edit Problem dialog (or via YAML configuration).
+     * 
+     * @param sandboxProgramName the name of the sandbox program to be used by this Problem, when sandbox usage is enabled.
+     */
+    public void setSandboxProgramName(String sandboxProgram) {
+        this.sandboxProgramName = sandboxProgram;
+    }
+
+    /**
+     * Returns the String containing the command used to invoke the sandbox configured for this problem.
+     * Note that the returned value is meaningless if the Problem has not been configured to use a sandbox. 
+     * 
+     * @return the command line used to invoke the sandbox for this problem, when sandbox usage is enabled.
+     */
+    public String getSandboxCmdLine() {
+        if (sandboxCmdLine == null) {
+            sandboxCmdLine = Constants.PC2_INTERNAL_SANDBOX_COMMAND_LINE;
```

Comment:
Same comment as for `getSandboxProgramName()`:  I don't think a getter should be causing a change in a Problem's configuration.

Reply:
Yep.  I still don't like having a getter with side effects... but maybe that needs to be handled as a separate Issue...